usaspending_api/core/validator/tinyshield.py

```diff
@@ -165,37 +185,56 @@ def block(self, request):
         base_minimum_fields = ('name', 'key', 'type')
 
-        for model in models:
-            if not all(field in model.keys() for field in base_minimum_fields):
-                raise Exception('Model {} missing a base required field [{}]'.format(model, base_minimum_fields))
+        if not all(field in model.keys() for field in base_minimum_fields):
+            raise Exception('Model {} missing a base required field [{}]'.format(model, base_minimum_fields))
 
-            if model['type'] not in VALIDATORS:
-                raise Exception('Invalid model type [{}] provided in description'.format(model['type']))
+        if model['type'] not in VALIDATORS:
+            raise Exception('Invalid model type [{}] provided in description'.format(model['type']))
 
-            type_description = VALIDATORS[model['type']]
-            required_fields = type_description['required_fields']
+        type_description = VALIDATORS[model['type']]
+        required_fields = type_description['required_fields']
 
-            for required_field in required_fields:
-                if required_field not in model:
-                    raise Exception('Model {} missing a type required field: {}'.format(model, required_field))
+        for required_field in required_fields:
+            if required_field not in model:
+                raise Exception('Model {} missing a type required field: {}'.format(model, required_field))
 
-            if model.get('text_type') and model['text_type'] not in SUPPORTED_TEXT_TYPES:
-                msg = 'Invalid model \'{key}\': \'{text_type}\' is not a valid text_type'.format(**model)
-                raise Exception(msg + ' Possible types: {}'.format(SUPPORTED_TEXT_TYPES))
+        if model.get('text_type') and model['text_type'] not in SUPPORTED_TEXT_TYPES:
+            msg = 'Invalid model \'{key}\': \'{text_type}\' is not a valid text_type'.format(**model)
+            raise Exception(msg + ' Possible types: {}'.format(SUPPORTED_TEXT_TYPES))
 
-            for default_key, default_value in type_description['defaults'].items():
-                model[default_key] = model.get(default_key, default_value)
+        for default_key, default_value in type_description['defaults'].items():
+            model[default_key] = model.get(default_key, default_value)
 
-            model['optional'] = model.get('optional', True)
+        model['optional'] = model.get('optional', True)
+
+        if model['type'] == 'any':
+            if in_any is True:
+                raise Exception('Nested "any" rules are not supported.')
+            # "any" child models are always optional and inherit their parent's name and key.
+            for sub_model in model['models']:
+                sub_model['name'] = model['name']
+                sub_model['key'] = model['key']
+                sub_model['optional'] = True
+                TinyShield.check_model(sub_model, True)
```

Comment:
What is the benefit of creating staticmethods and calling a new class instead of using a class method and `self`?

Reply:
No benefit.  Probably just an artifact of refactoring.  Good catch.